### ISSUE TITLE
Support for drawing to multiple rendertargets

### ISSUE THREAD

**Issue by Jhonnyg** [defold/defold#6628]
**Is your feature request related to a problem? Please describe (REQUIRED):**
Several graphical effects (e.g Many-light pipelines / deferred rendering, ssao, temporal antialiasing) operate on several components of a frame such as position, normals and albedo color. These are usually generated by drawing the scene once into several render targets that can be used in other parts of the rendering pipeline. MRT support is available from OpenGL 3 / ES3 / WebGL2 (and baseline vulkan), but is currently not exposed in any way.

**Describe the solution you'd like (REQUIRED):**
When creating a rendertarget in a render script, we could pass in extra params for the additional targets we wish to allocate:

Current API:
```
local color_params = { ... }
local depth_params = { ... }

self.my_render_target = render.render_target({
  [render.BUFFER_COLOR_BIT] = color_params,
  [render.BUFFER_DEPTH_BIT] = depth_params })`
```

**API idea 1:**

```
self.my_render_target = render.render_target({
	[render.BUFFER_COLOR_BIT_0] = color_params,
	[render.BUFFER_COLOR_BIT_1] = color_params,
	[render.BUFFER_COLOR_BIT_2] = color_params_with_float_type,
	[render.BUFFER_COLOR_BIT_3] = color_params_with_other_filtering,
	[render.BUFFER_DEPTH_BIT] = depth_params
})
```

**API idea 2:**

```
local color_params = {
	{...}, -- properties for buffer 1
	{...}, -- properties for buffer 2
	{...}, -- properties for buffer 3
	etc
}

self.my_render_target = render.render_target({
	[render.BUFFER_COLOR_BIT] = color_params,
	[render.BUFFER_DEPTH_BIT] = depth_params
})
```

**Additional context (OPTIONAL):**

Open questions:
OpenGL es2 / webgl 1 does support this functionality, but only through certain extensions which might not be universally supported. Right now we don't have a way to detect what extensions or features that are available outside the vanilla API, so for those graphics contexts, this would just not work. Do we want to have some way of detecting certain non-standard "features" in the engine somehow? If a certain platform doesn't support MRTs, a user might want to turn off certain effects, or do a workaround. Maybe something like render.has_support(render.MULTI_RENDERTARGET) or render.enable(render.MULTI_RENDERTARGET) that
a developer can use to decide how the render scripts should be configured.

**Comment by britzl** [comment]
> Do we want to have some way of detecting certain non-standard "features" in the engine somehow? If a certain platform doesn't support MRTs, a user might want to turn off certain effects, or do a workaround. Maybe something like render.has_support(render.MULTI_RENDERTARGET) or render.enable(render.MULTI_RENDERTARGET) that
> a developer can use to decide how the render scripts should be configured

Related: https://github.com/defold/defold/issues/3350

Another thing to note is that we do conditionally make some features available already. The floating point render target constants, for example render.FORMAT_RGB16F, are not set if they are not available. See release notes for 1.3.1: https://forum.defold.com/t/defold-1-3-1-has-been-released/70673

**Comment by JCash** [comment]
As for the api, I lean more towards #1, to be more explicit in the call, matching enum with properties.

**Comment by Jhonnyg** [comment]
> As for the api, I lean more towards #1, to be more explicit in the call, matching enum with properties.

Yeah, I kinda like that one as well. I'll use that as starting point of the feature when I start making it 👍

**Comment by britzl** [comment]
There seems to be a problem on Android still:

https://forum.defold.com/t/defold-1-3-5-beta/71292/22?u=britzl

**Comment by Grify** [comment]
Why is this still open? Didn't 1.3.5 already implement this?

**Comment by Jhonnyg** [comment]
> Why is this still open? Didn't 1.3.5 already implement this?

There was an issue with Android that now has been fixed

**Comment by Jhonnyg** [comment]
Fixed by #6844
